### PR TITLE
13363 - buildFile renamed to buildFile.xml for 3.5 and beyond

### PR DESCRIPTION
**Pull request by Cattlesquat**
Now featuring a file extension for Extra Sanity!

* Looks for "buildFile" for all versions 3.4.999 and earlier. 
* Looks for "buildFile.xml" for all versions after that
* Writes buildFile.xml
* And then removes old "buildFile" instances lingering in the archive

### CODE REVIEW

File: vassal-app/src/main/java/VASSAL/build/GameModule.java [hunk 1]
```diff
@@ -337,18 +341,29 @@ protected void build() throws IOException {
       build(null);
     }
     else {
+      final AbstractMetaData data = MetaDataFactory.buildMetaData(f);
+      final String properBuildFileName = (VersionUtils.compareVersions(data.getVassalVersion(), "3.4.999") < 0) ? BUILDFILE_OLD : BUILDFILE;
```
Comment (uckelman) [inline vassal-app/src/main/java/VASSAL/build/GameModule.java:345]: Is there a reason you can't do `VersionUtils.compareVersions(data.getVassalVersion(), "3.5")`? That ought to be negative for anything 3.4 or earlier.

File: vassal-app/src/main/java/VASSAL/build/module/ModuleExtension.java [hunk 2]
```diff
@@ -116,7 +118,14 @@ public boolean getUniversal() {
   }
 
   public void build() {
-    final String fileName = GameModule.BUILDFILE;
+
+    final AbstractMetaData data = MetaDataFactory.buildMetaData(archive.getArchive().getFile());
+    final String fileName = (VersionUtils.compareVersions(data.getVassalVersion(), "3.4.999") < 0) ? GameModule.BUILDFILE_OLD : GameModule.BUILDFILE;
```
Comment (uckelman) [inline vassal-app/src/main/java/VASSAL/build/module/ModuleExtension.java:123]: Ditto.

File: vassal-app/src/main/java/VASSAL/tools/ArchiveWriter.java [hunk 3]
```diff
@@ -197,6 +197,20 @@ public void removeImage(String name) {
...
+   * @param name file in the archive to be removed
+   */
+  public void removeFile(String name) {
+    try {
+      archive.remove (name);
+    }
+    catch (IOException e) {
+      WriteErrorDialog.error(e, archive.getName());
+    }
+  }
```
Comment (uckelman) [inline vassal-app/src/main/java/VASSAL/tools/ArchiveWriter.java:211]: While you're at it, you could replace all but the last line of `removeImage()` with a call to `removeFile(imageDir + name)`.

**Review by uckelman — CHANGES_REQUESTED** [review]

File: vassal-app/src/main/java/VASSAL/build/GameModule.java (hunk 1 above)
Comment (Cattlesquat) [reply vassal-app/src/main/java/VASSAL/build/GameModule.java:345]: That's actually what I had in the original version, I think, and then 3.5.0-SNAPSHOT turned out to be earlier?

Comment (Cattlesquat) [reply vassal-app/src/main/java/VASSAL/build/GameModule.java:345]: I need a version that's either "guaranteed lower than ANY OTHER 3.5" or "guaranteed higher than ANY OTHER 3.4".

File: vassal-app/src/main/java/VASSAL/tools/ArchiveWriter.java (hunk 3 above)
Comment (uckelman) [reply vassal-app/src/main/java/VASSAL/tools/ArchiveWriter.java:204]: _all but the last line_. You killed the last line.

File: vassal-app/src/main/java/VASSAL/build/GameModule.java (hunk 1 above)
Comment (uckelman) [reply vassal-app/src/main/java/VASSAL/build/GameModule.java:345]: Ah, crud, right. Then you want to use `VersionUtils.trunctateToMinorVersion()` on `data.getVassalVersion()` first, to cut off anything past the minor version.

**Review by uckelman — APPROVED** [review]

**Pull request merged**